src/main/java/seedu/badMaths/TrigoGraph.java

```diff
@@ -48,17 +72,48 @@ public void splitTrigoIntoPhasors(String trigo) {
     }
 
     public void splitPhasorsIntoFreq(String phasors) {
-        String[] freqAndShift = phasors.split("\\+", 2);
+        String[] freqAndShift = new String[2];
+        boolean is_negative = false;
+        if (phasors.contains("+")) {
+            freqAndShift = phasors.split("\\+", 2);
+        } else {
+            freqAndShift = phasors.split("-", 2);
+            is_negative = true;
+        }
+
         String freqWithX = freqAndShift[0];
-        String[] freqComponents = freqWithX.split("\\*", 3);
-        Double freq;
-        if (trigoEqn.contains("pi")) {
-            freq = Double.parseDouble(freqComponents[0]) / 2;
+        String[] freqComponents = new String[1];
+        if (freqWithX.equals("x")) {
+            freqComponents[0] = "1.0";
         } else {
-            freq = Double.parseDouble(freqComponents[0]) / (2 * Math.PI);
+            freqComponents = freqWithX.split("\\*", 2);
+        }
+        Double freq;
+        try {
+            if(freqAndShift[1].contains("+")){
+                throw new NumberFormatException();
+            }
+            if(freqComponents[1].equals("pi*x") || freqComponents[1].equals("x")) {
+                if (trigoEqn.contains("pi")) {
+                    freq = Double.parseDouble(freqComponents[0]) / 2;
+                } else {
+                    freq = Double.parseDouble(freqComponents[0]) / (2 * Math.PI);
+                }
```

Comment:
For this if statement, trigoEqn.contains("pi") is always true if freqComponents[1].equals(pi*x). So is it better to just write:
if(freqComponents[1].equals("pi*x") {
       freq = Double.parseDouble...
} ....

Looks cleaner if we split the if statements up

Reply:
It's to catch for cases when user write 2*x and 2*pi*x which are both valid cases